FindPETSc.cmake

```diff
@@ -47,9 +47,18 @@ else()
   endforeach()
 endif()
 
+macro (check_if_arch dir)
+  set (PETSC_TMP "${PETSC_DIR}/")
+  if (EXISTS "${PETSC_DIR}/${PETSC_ARCH}/${dir}")
+    set (PETSC_TMP "${PETSC_TMP}/${PETSC_ARCH}")
```

Comment:
Why do you need this? If `PETSC_ARCH` is unset and there is no subdirectory then the old version just works. If `PETSC_ARCH` is explicitly set to something that does not exist (or is corrupt) then we _should_ get an error instead of succeeding with no `PETSC_ARCH`.